### PR TITLE
Re-render JsonEditor on value input

### PR DESCRIPTION
**Pull request by fiskus**
Reduce React state usage inside JsonEditor. Make external value as a source of truth.

JsonEditor still has an internal state: sortOrder for keys and fieldsPath (it's like URL for the entire component), in other aspects internal code become clearer, I hope.

### CODE REVIEW

File: catalog/app/utils/json-schema/json-schema.js [hunk 1]
```diff
@@ -126,7 +126,7 @@ export function makeSchemaDefaultsSetter(optSchema) {
     const validate = ajv.compile(schema)
 
     return (obj) => {
-      validate(obj)
+      validate(R.clone(obj))
```
Comment (nl0) [inline catalog/app/utils/json-schema/json-schema.js:129]: i thought the point of this call is to mutate the object and set the defaults -- in this case i believe you should return the cloned object

**Review by nl0 — APPROVED** [review]
seems good, some inline comments

File: catalog/app/utils/json-schema/json-schema.js (hunk 1 above)
Comment (fiskus) [reply catalog/app/utils/json-schema/json-schema.js:129]: I think, I made a mistake, will investigate it. Thank you

File: catalog/app/components/JsonEditor/State.js [hunk 2]
```diff
@@ -108,14 +86,15 @@ export function iterateSchema(schema, sortCounter, parentPath, memo) {
       key,
       parentPath,
       required,
-      sortIndex: sortCounter.current,
+
```
[the remark this answers is not in the document]
Comment (fiskus) [reply catalog/app/components/JsonEditor/State.js:89]: It was :) It helped me refactor, I will remove it

**Pull request merged**